ntoskrnl/se/token.c

```diff
@@ -2102,9 +2106,13 @@ SepPerformTokenFiltering(
     BOOLEAN FoundGroup;
     PAGED_CODE();
 
-    /* Ensure that the token we get is not garbage */
+    /* Ensure that the token we get is valid */
     ASSERT(Token);
 
+    // FIXME: This function **MUST** exclusively lock the source token,
+    // instead of its callers!! (see SeFilterToken missing lock, while
+    // NtFilterLock does the lock too early)
+
```

Comment:
Regarding that, `SepAcquireTokenLockExclusive(Token);` could be moved here from `NtFilterToken` since we're pretty much capturing data from the caller and not the token anyway.

Reply:
exactly!